src/typing.ml

```diff
@@ -684,7 +687,7 @@ let type_type_declaration kid crcm ns tdl =
   tdl
 
 let process_sig_type ~loc ?(ghost = Nonghost) kid crcm ns r tdl =
-  let tdl = type_type_declaration kid crcm ns tdl in
+  let tdl = type_type_declaration kid crcm ns r tdl in
```

Comment:
Why don't you use `Tast.rec_flag` here (with the `rec_flag` converter used the line below)?